File: docs/snippets/gna/import_export.py
# Copyright (C) 2022 Intel Corporation
# SPDX-License-Identifier: Apache-2.0

#! [import]
import openvino as ov
from io import BytesIO
#! [import]

from snippets import get_model

def main():
    model = get_model()
    blob_path = "compiled_model.blob"

    core = ov.Core()
    if "GNA" not in core.available_devices:
        return 0

    compiled_model = core.compile_model(model, device_name="GNA")

    #! [ov_gna_export]
    user_stream = compiled_model.export_model()
    with open(blob_path, "wb") as f:
        f.write(user_stream)
    #! [ov_gna_export]

    # [ov_gna_import]
    with open(blob_path, "rb") as f:
        buf = BytesIO(f.read())
        compiled_model = core.import_model(buf, device_name="GNA")
    # [ov_gna_import]

File: docs/snippets/gna/import_export.cpp
// Copyright (C) 2022 Intel Corporation
// SPDX-License-Identifier: Apache-2.0
//
//! [include]
#include <fstream>
#include <openvino/openvino.hpp>
//! [include]

int main() {
const std::string model_path = "model.xml";
const std::string blob_path = "compiled_model.blob";

ov::Core core;
auto model = core.read_model(model_path);
auto compiled_model = core.compile_model(model, "GNA");

{
//! [ov_gna_export]
std::ofstream ofs(blob_path, std::ios_base::binary | std::ios::out);
compiled_model.export_model(ofs);
//! [ov_gna_export]
}
{
//! [ov_gna_import]
std::ifstream ifs(blob_path, std::ios_base::binary | std::ios_base::in);
auto compiled_model = core.import_model(ifs, "GNA");
//! [ov_gna_import]
}
}